node-notes-cli/main.ts

```diff
@@ -0,0 +1,75 @@
+import * as fs from 'node:fs/promises';
+
+type Data = {
+  nextId: number;
+  notes: Record<string, string>;
+};
+
+let d: string;
```

Comment:
You don't reference this anywhere except down below at line 45, so it would be best to declare it down there, on line 44.